fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushdownFilterThroughWindow.java

```diff
@@ -0,0 +1,204 @@
+ * The following will demonstrate how the plan changes:
+ * Logical plan tree:
+ *                 any_node
+ *                   |
+ *                filter (row_number <= 100)
+ *                   |
+ *                window (PARTITION BY a ORDER BY b)
+ *                   |
+ *                 any_node
+ * transformed to:
+ *                 any_node
+ *                   |
+ *                filter (row_number <= 100)
+ *                   |
+ *                window (PARTITION BY a ORDER BY b)
+ *                   |
+ *                partition_topn(PARTITION BY: a, ORDER BY b, Partition Limit: 100)
+ *                   |
+ *                 any_node
+ */
+
+public class PushdownFilterThroughWindow extends OneRewriteRuleFactory {
+
+    @Override
+    public Rule build() {
+        return logicalFilter(logicalWindow()).then(filter -> {
+            LogicalWindow<Plan> window = filter.child();
+
+            // We have already done such optimization rule, so just ignore it.
+            if (window.child(0) instanceof LogicalPartitionTopN) {
+                return filter;
+            }
+
+            // Check the window function. There are some restrictions for window function:
+            // * The number of window function should be 1.
+            // * The window function should be one of the 'row_number()', 'rank()', 'dense_rank()'.
+            // * The window type should be 'ROW'.
+            // * The window frame should be 'UNBOUNDED' to 'CURRENT'.
+            // * The 'PARTITION' key and 'ORDER' key can not be empty at the same time.
+            List<NamedExpression> windowExprs = window.getWindowExpressions();
+            if (windowExprs.size() != 1) {
+                return filter;
+            }
+            NamedExpression windowExpr = windowExprs.get(0);
+            if (windowExpr.children().size() != 1 || !(windowExpr.child(0) instanceof WindowExpression)) {
+                return filter;
+            }
+
+            WindowExpression windowFunc = (WindowExpression) windowExpr.child(0);
+            // Check the window function name.
+            if (!LogicalWindow.checkWindowFuncName4PartitionTopN(windowFunc)) {
+                return filter;
+            }
+
+            // Check the partition key and order key.
+            if (!LogicalWindow.checkWindowPartitionAndOrderKey4PartitionTopN(windowFunc)) {
+                return filter;
+            }
+
+            // Check the window type and window frame.
+            if (!LogicalWindow.checkWindowFrame4PartitionTopN(windowFunc)) {
+                return filter;
+            }
+
+            // Check the filter conditions. Now, we currently only support simple conditions of the form
+            // 'column </ <=/ = constant'. We will extract some related conjuncts and do some check.
+            Set<Expression> conjuncts = filter.getConjuncts();
+            boolean existsORConditionsInConjuncts = conjuncts.stream().anyMatch(this::existOR);
```

Comment:
why check this? i think we could push any one of conjunct in this set

Reply:
Remove it. I got some misunderstanding.